crates/stc_ts_type_checker/tests/conformance/types/conditional/conditionalTypes2.stats.rust-debug

```diff
@@ -1,6 +1,6 @@
 Stats {
     required_error: 2,
     matched_error: 5,
-    extra_error: 5,
+    extra_error: 6,
```

Comment:
This regression is due to previous behavior

Reply:
We returned `any` so errors were not validated correctly

Reply:
I extracted this PR from https://github.com/dudykr/stc/pull/686 because of this regression